Bio/SeqUtils/MeltingTemp.py

```diff
@@ -776,19 +776,21 @@ def Tm_GC(
     seq = str(seq)
     if check:
         seq = _check(seq, "Tm_GC")
-    percent_gc = SeqUtils.GC(seq)
-    # Ambiguous bases: add 0.5, 0.67 or 0.33% depending on G+C probability:
-    tmp = (
-        sum(map(seq.count, ("K", "M", "N", "R", "Y"))) * 50.0 / len(seq)
-        + sum(map(seq.count, ("B", "V"))) * 66.67 / len(seq)
-        + sum(map(seq.count, ("D", "H"))) * 33.33 / len(seq)
-    )
+
+    tmp = any([(x in seq) for x in "KMNRYBVDH"])
```

Comment:
I would use one line here, combine with the if-statement. Also you don't need a list comprehension at all (and some flake8 tools will complain about this redundancy). Perhaps:

```python
if strict and any(x in seq for x in "KMNRYBVDH"):
```

Or we could use set intersection but that seems overkill.

[This isn't your assumption, but can we assume the sequence is upper case? Perhaps that should be done earlier in the function?]

Reply:
You are right, will change to the suggested version.

About the uppercase question, in the lines above we have:

```
if check:
        seq = _check(seq, "Tm_GC")
```

`_check` returns an uppercase string, and as far as I understand if the user passes `check=False` in the arguments we can also assume that the sequence is already in the correct format.